jdbc/src/main/java/org/openconnectors/JdbcConfigKeys.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.openconnectors;
+
+public class JdbcConfigKeys {
+    /**
+     * JDBC connection URL
+     */
+    public static final String CONNECTION_URL_CONFIG = "connection.url";
+
+    /**
+     * Database user
+     */
+    public static final String CONNECTION_USER_CONFIG = "connection.user";
+
+    /**
+     * Database password
+     */
+    public static final String CONNECTION_PASSWORD_CONFIG = "connection.password";
+
+    /**
+     * Maximum number of retry to connect to the database.
+     */
+     public static final String CONNECTION_MAX_ATTEMPT = "connection.max.attempt";
+
+     /**
+     * Delay between retries to connect to the database.
+     */
+    public static final String CONNECTION_RETRY_DELAY = "connection.retry.delay";
+
+    /**
+     * Schema pattern.
+     */
+    public static final String SCHEMA_PATTERN = "schema.pattern";
+
+    /**
+     * Mode of querying. Can be `mode.bulk`, `mode.autoincrementing`, `mode.incrementing`, `mode.by_query`.
```

Comment:
So this config key needs to be set to the string value? What about the keys below - should they be set to boolean for on/off?

Reply:
BULK_MODE and INCREMENTING_MODE should not be set at all. This is just a constants for MODE configuring. So, there are 2 options: MODE = mode.bulk(which is the value of BULK_MODE) or MODE = mode.incrementing(which is the value of INCREMENTING_MODE constant).